chore: fix dependabot

bigstack-brian-su opened the pull request:
- add schedule scanning to monday 9am
- group pull request to reduce fatigue
- set pull request sensitivity

#### What type of PR is this?

Fix

#### What this PR does / why we need it

#### Which issue(s) this PR fixes

Fixes #

#### Special notes for your reviewer

> Note

#### Additional documentation

```docs

```

bigstack-brian-su committed:
chore: fix dependabot

- add schedule scanning to monday 9am
- group pull request to reduce fatigue
- set pull request sensitivity

Signed-off-by: Brian Su <brian.su@bigstack.co>

bigstack-brian-su committed:
chore: remove redundant configuration

Signed-off-by: Brian Su <brian.su@bigstack.co>

steven-chiu-bigstack reviewed (approved):
Hi @bigstack-brian-su 

The format and configuration look good based on the [Dependabot docs](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/dependabot-options-reference), thanks for taking care of this! 🙏 

Since we don't have unit tests and E2E tests at the moment, dependency updates could potentially break functionality without anyone knowing it, though this is less likely for minor and patch updates. However, it's not quite possible to implement E2E tests for all pages considering the time it takes, so it's probably better to do necessary updates (e.g., security updates) as early as possible before the application grows even larger.

For the `security-updates` group, I think it makes sense to always update those packages as recommended, as they typically don’t affect functionality or change library interfaces.

But I'm not quite sure what is the right thing to do with the `version-updates` group. We could either go with the current setup and close the version-updates PR every week or review the changelogs for each package and upgrade only if there's no major conerns.

Simply put, in my opinion, the simplest solution for us at the current stage might be:

1. Ignore `version-updates` by closing the weekly PR or commenting them out in this config file until E2E coverage improves, and
2. Apply all `security-updates`.

Does that make sense? 🤔 @bigstack-brian-su  @raven-pan @lizzy-liang-bigstack

bigstack-brian-su commented:
> Hi @bigstack-brian-su
> 
> The format and configuration look good based on the [Dependabot docs](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/dependabot-options-reference), thanks for taking care of this! 🙏
> 
> Since we don't have unit tests and E2E tests at the moment, dependency updates could potentially break functionality without anyone knowing it, though this is less likely for minor and patch updates. However, it's not quite possible to implement E2E tests for all pages considering the time it takes, so it's probably better to do necessary updates (e.g., security updates) as early as possible before the application grows even larger.
> 
> For the `security-updates` group, I think it makes sense to always update those packages as recommended, as they typically don’t affect functionality or change library interfaces.
> 
> But I'm not quite sure what is the right thing to do with the `version-updates` group. We could either go with the current setup and close the version-updates PR every week or review the changelogs for each package and upgrade only if there's no major conerns.
> 
> Simply put, in my opinion, the simplest solution for us at the current stage might be:
> 
> 1. Ignore `version-updates` by closing the weekly PR or commenting them out in this config file until E2E coverage improves, and
> 2. Apply all `security-updates`.
> 
> Does that make sense? 🤔 @bigstack-brian-su @raven-pan @lizzy-liang-bigstack

Yes, let me adjust it to not open any pull requests for `version-updates` but it will still generate the alerts.

bigstack-brian-su committed:
chore: update dependabot configuration and dep review settings

Update configuration as per discussion in pull request, use build in
option to only retain grouped security updates.

Reduce dependency review information to only when failures occur.

Signed-off-by: Brian Su <brian.su@bigstack.co>

abemscac commented:
Thanks @bigstack-brian-su! I think we can merge this PR and see how it goes. 🙏
